src/rgw/rgw_kms.cc

```diff
@@ -85,6 +85,67 @@ class VaultSecretEngine: public SecretEngine {
     return res;
   }
 
+  int load_token_from_serviceaccount(std::string *vault_token)
+  {
+    int res = 0;
+    bufferlist vault_bl;
+    std::string vault_url = cct->_conf->rgw_crypt_vault_addr;
+    if (vault_url.empty()) {
+      ldout(cct, 10) << "ERROR: Vault address not set in rgw_crypt_vault_addr" << dendl;
+      return -EINVAL;
+    }
+
+    //Reading jwt token for service account token file
+    char buf[2048];
+    res = safe_read_file("", "/var/run/secrets/kubernetes.io/serviceaccount/token", buf, sizeof(buf));
```

Comment:
where does this path come from? is it sensible to hard-code it here, or should it be configurable?

Reply:
indeed, it should

Reply:
> is this something we can test in teuthology without actually setting up rook/kubernetes? or maybe the test coverage belongs in rook itself?

IMO the test coverage belongs to rook itself. It requires k8s cluster the test it ou(authentication is specific to a service account in k8s)t. Usually in k8s RGW is setup via Rook. I am not sure whether it can be done in teuthology tests

Reply:
> 

The token for the default service accounts for pod will be stored here, simple googling didn't give me any option to change the this path. @leseb any ideas.

Reply:
Right, but nothing about the internal logic of this PR is specific to k8s.  Casey is pointing out that we don't hard code paths in RGW source code, in general.  So this seems to belong in a config option.

Reply:
It is possible to use "kubectl create serviceaccount" to create additional bearer tokens.  From what I can tell these should be functionally very similar to the default service account - but you would obviously need to store the token somewhere different.  So yes, yes, this absolutely should be configurable.

Reply:
@mattbenjamin @mdw-at-linuxbox @cbodley, Okay I understood, is it okay to use existing token file option(`rgw_crypt_vault_token_file`) for this than creating new option?

Reply:
I believe you'll want to use a different name for 3 reasons: /1/ you can't intelligently default it if you overload it this way, /2/ you'll have to explain in the documentation how it could hold several different things - /3/ this creates the need for logic to intelligently complain in case the user leaves this pointing at the wrong kind of token.  (you think people read documentation?  Hah!)

Reply:
Agreed.